frontend/app/src/components/StreamlitDialog/DeployDialog/DeployModal.tsx

```diff
@@ -14,24 +14,54 @@
  * limitations under the License.
  */
 
-import React, { ReactElement } from "react"
+import React, { ReactElement, ReactNode } from "react"
 
+import { ModalBody as UIModalBody } from "baseui/modal"
 import { CloseSource } from "baseui/modal/types"
+import { useTheme } from "@emotion/react"
 
-import { Modal, ModalBody, ModalHeader } from "@streamlit/lib"
+import { EmotionTheme, Modal, ModalHeader } from "@streamlit/lib"
 
 interface IDeployModalProps {
   children: React.ReactNode
   onClose: (a: { closeSource?: CloseSource }) => unknown
 }
 
+export interface ModalBodyProps {
+  children: ReactNode
+}
+
+function ModalBody({ children }: Readonly<ModalBodyProps>): ReactElement {
+  const { colors, fontSizes, spacing }: EmotionTheme = useTheme()
+
+  return (
+    <UIModalBody
+      style={{
+        marginTop: spacing.none,
+        marginLeft: spacing.none,
+        marginRight: spacing.none,
+        marginBottom: spacing.none,
+        paddingTop: spacing.md,
+        paddingRight: spacing.none,
+        paddingBottom: spacing.none,
+        paddingLeft: spacing.none,
+        color: colors.bodyText,
+        fontSize: fontSizes.md,
+        overflowY: "auto",
+      }}
+    >
+      {children}
+    </UIModalBody>
+  )
+}
+
 function DeployModal(
   props: React.PropsWithChildren<IDeployModalProps>
 ): ReactElement {
   const { children, onClose } = props
   return (
-    <Modal isOpen={true} closeable={true} onClose={onClose} size="full">
-      <ModalHeader>Deploy this app</ModalHeader>
+    <Modal isOpen={true} closeable={true} onClose={onClose} size={964}>
```

Comment:
nit: we are currently preparing for advanced theming where customers are going to be able to set the base font size and similar, which is why we tried to get rid of hardcoded numbers in styles. It'd be great if there were a way to derive the modal's width automatically or set a max width and let the columns define themselves based on the content instead of hardcoding a number here. Because I could imagine that as soon as customers define different sizes, this is going to look awkward. Also it might not be responsive at all when the number is hardcoded in pixels (although to be fair, I am not sure how responsive it is right now).

If automatically deriving it does not work, perhaps putting a new number into `spacing` and using a `rem` unit to make it a little bit more relative to the root might work.

Reply:
for context: we discussed to change it to `"auto"` and determine the width based on the `StyledCardContainer` to not impact the other dialogs.